homeassistant/components/media_player/pjlink.py

```diff
@@ -94,14 +94,19 @@ def projector(self):
     def update(self):
         """Get the latest state from the device."""
         with self.projector() as projector:
-            pwstate = projector.get_power()
-            if pwstate == 'off':
+            try:
+                pwstate = projector.get_power()
+                if pwstate == 'on' or pwstate == 'warm-up':
+                    self._pwstate = STATE_ON
+                else:
+                    self._pwstate = STATE_OFF
+                self._muted = projector.get_mute()[1]
+                self._current_source = \
+                    format_input_source(*projector.get_input())
+            except:
```

Comment:
Make sure to only catch exceptions we expect.

Reply:
I could make the changes you mentioned, but I'm new to Github and don't know the processes yet. Do I have to open a new pull request or is it possible to modify the existing pull request?